nixos/modules/services/networking/consul.nix

```diff
@@ -148,107 +107,61 @@ in
           type = types.bool;
         };
       };
-
     };
-
   };
 
   config = mkIf cfg.enable (
-    mkMerge [{
-
-      users.users.consul = {
-        description = "Consul agent daemon user";
-        uid = config.ids.uids.consul;
-        # The shell is needed for health checks
-        shell = "/run/current-system/sw/bin/bash";
-      };
-
-      environment = {
-        etc."consul.json".text = builtins.toJSON configOptions;
-        # We need consul.d to exist for consul to start
-        etc."consul.d/dummy.json".text = "{ }";
-        systemPackages = [ cfg.package ];
-      };
-
-      systemd.services.consul = {
-        wantedBy = [ "multi-user.target" ];
-        after = [ "network.target" ] ++ systemdDevices;
```

Comment:
> this service was previously discovering host IP addresses in an unnecessarily complex way, when Consul supports using [`go-sockaddr`](https://github.com/hashicorp/go-sockaddr) templates in configuration

It's unclear to me whether `go-sockaddr` does the equivalent of the removed `++ systemdDevices`, does anyone know more?